### DISCUSSION TITLE
Where is vercel.json redirect and rewrite syntax documentation?

### DISCUSSION THREAD

**Discussion by trusktr** [vercel/community#1367] [Help]
I see both docs:

- https://vercel.com/docs/project-configuration#project-configuration/redirects
- https://vercel.com/docs/project-configuration#project-configuration/rewrites

Neither of them includes documentation on the syntax for `source` and `destination`, only a few examples. It's not obvious at all how the syntax works from those docs.

EDIT: I found the [rewrite docs for Next.js](https://nextjs.org/docs/api-reference/next.config.js/rewrites) (I'm not using Next.js, so I would never think to look there).

It _seems_ that Next.js docs also the documentation for the same syntax for plain Vercel, but that's only a *guess*, because nowhere in the Vercel docs does it tell us to look at Next.js for further guidance on syntax.

---

The problem I have, for example, is that the mapping I have, `{"source": "/:path*(\\.md)", "destination": "/:path*"},`, seems to not work. Apart from going to dig into the source code (which I really want to avoid), I can't see a clue anywhere that indicates if this is actually correct syntax.

**Comment by amyegan** [comment]
This is a great callout @trusktr. I'm going to open a new ticket to update the documentation on rewrite/redirect source syntax.

Until the change is made, you can find examples in the [Next.js rewrite docs](https://nextjs.org/docs/api-reference/next.config.js/rewrites) that you found and in the [Path-to-RegExp](https://github.com/pillarjs/path-to-regexp) readme
